src/sentry/search/eap/spans/formulas.py

```diff
@@ -0,0 +1,90 @@
+from collections.abc import Callable
+from typing import Any
+
+from sentry_protos.snuba.v1.attribute_conditional_aggregation_pb2 import (
+    AttributeConditionalAggregation,
+)
+from sentry_protos.snuba.v1.endpoint_trace_item_table_pb2 import Column
+from sentry_protos.snuba.v1.trace_item_attribute_pb2 import (
+    AttributeKey,
+    AttributeValue,
+    ExtrapolationMode,
+    Function,
+    StrArray,
+)
+from sentry_protos.snuba.v1.trace_item_filter_pb2 import ComparisonFilter, TraceItemFilter
+
+from sentry.exceptions import InvalidSearchQuery
+from sentry.search.eap.columns import ArgumentDefinition, FormulaDefinition
+from sentry.search.eap.constants import RESPONSE_CODE_MAP
+
+
+def literal_validator(values: list[Any]) -> Callable[[str], bool]:
+    def _validator(input: str) -> bool:
+        if input in values:
+            return True
+        raise InvalidSearchQuery(f"Invalid parameter {input}. Must be one of {values}")
+
+    return _validator
```

Comment:
I should've commented this last time sorry, lets move this validator and the count_processor to `eap/utils.py` so they can be shared

Reply:
👍 I forgot about the validator, i'll move it